src/trio/_tests/test_testing_raisesgroup.py

```diff
@@ -125,6 +185,37 @@ def test_check() -> None:
             raise ExceptionGroup("", (ValueError(),))
 
 
+def test_unwrapped_match_check() -> None:
+    def my_check(e: object) -> bool:  # pragma: no cover
+        return True
+
+    msg = (
+        "`allow_unwrapped=True` bypasses the `match` and `check` parameters"
+        " if the exception is unwrapped. If you intended to match/check the"
+        " exception you should use a `Matcher` object. If you want to match/check"
+        " the exceptiongroup when the exception *is* wrapped you need to"
+        " do e.g. `if isinstance(exc.value, ExceptionGroup):"
+        " assert RaisesGroup(...).matches(exc.value)` afterwards."
+    )
```

Comment:
Do we do this elsewhere? I'm pretty sure that elsewhere we try to catch only snippets of the message.

Though that's probably more because the message elsewhere is variable. I'm a bit concerned that this is brittle though I'm sure that if we update the message to be more descriptive the person doing that will be fine copying it here.

Reply:
nah it's not them being variable, the match strings in `test_catch_unwrapped_exceptions` was mostly me being lazy - but also because in all those tests the error message in question is only matched against once, whereas here it's matched for twice.
It was a bit of a pain when iterating the message, but any future updates to the message should indeed be simply copy-paste jobs to update the string in the test.

Quoting the whole string also does have the positive side effect of effectively being a comment explaining the test case.